pkg/experimental/pkg_filegroup.bzl

```diff
@@ -578,6 +591,83 @@ pkg_mkdirs = rule(
             The default is `dir`.
             """,
             default = "dir",
+            # TODO(nacl): use "values" here
+        ),
+    },
+)
+
+def _pkg_mklinks_impl(ctx):
+    _validate_attr(ctx.attr.attrs)
+    return [
+        PackageSymlinkInfo(
+            link_map = ctx.attr.links,
+            attrs = ctx.attr.attrs,
+            section = ctx.attr.section,
+        ),
+    ]
+
+pkg_mklinks = rule(
+    doc = """pkg_filegroup-like-rule for symlink management.
+
+    This rule results in the creation of one or more symbolic links a package.
+    The link may point to a location outside of it.
+
+    """,
+    implementation = _pkg_mklinks_impl,
+    attrs = {
+        "links": attr.string_dict(
+            doc = """Link mappings to create.
+
+            The keys of this dict are the link names, the values are the source
+            files.
+
+            Note that relative paths are important here: the package builder MAY
+            adjust the "link name" of the pair (i.e. make paths absolute), but
+            it MUST NOT attempt to do anything with the "source" part.
+
+            """,
+            mandatory = True,
+        ),
+        "attrs": attr.string_list_dict(
+            doc = """Attributes to set for the output targets.
+
+            Must be a dict of:
+
+            ```
+            "unix" : [
+                "Four-digit octal permissions string (e.g. "0755") or "-" (don't change from what's provided),
+                "User Id, or "-" (use current user)",
+                "Group Id, or "-" (use current group)",
+            ]
+            ```
+
+            The permissions value defaults to "0777".  The user/group values
+            default to "-".
+            """,
+            default = {"unix": ["0777", "-", "-"]},
```

Comment:
So what will happen on other OSes?  Random values?

Reply:
The theory is that packager implementation can chose to interpret permission values as they wish.  Tarfiles on Windows, for example, will probably still want to use UNIX-style filesystem permissions.

The `unix` permissions mapping was intended entirely to support the common UNIX filesystem packaging use case (tars, debs, rpms).  Other keys to the dictionary could theoretically be added to support additional permissions types (UNIX extended attributes, Windows permissions, etc.).

There probably is a way to make this look prettier, but I haven't given it much thought.